iodata/formats/mwfn.py

```diff
@@ -0,0 +1,367 @@
+            'nbasis': nbasis, 'nindbasis': nindbasis, 'nprims': nprim,
+            'nshells': nshell, 'nprimshells': nprimshell, 'full_virial_ratio': vt_ratio,
+            'shell_centers': shell_centers, 'shell_types': shell_types,
+            'prim_per_shell': prim_per_shell, 'exponents': exponent, 'coeffs': coeffs,
+            'mo_numbers': mo_numbers, 'mo_occs': mo_occs, 'mo_energies': mo_energies,
+            'mo_coeffs': mo_coeffs, 'mo_type': mo_type, 'mo_sym': mo_sym}
+
+
+def build_obasis(shell_map: np.ndarray, shell_types: np.ndarray,
+                 exponents: np.ndarray, prim_per_shell: np.ndarray,
+                 coeffs: np.ndarray,
+                 ) -> Tuple[MolecularBasis]:
+    """Based on the fchk modules basis building.
+
+    Parameters
+    -------------
+    shell_map:  np.ndarray (integer)
+        Index of what atom the shell is centered on. The mwfn file refers to this section
+        as `Shell centers`. Mwfn indices start at 1, this has been modified and starts
+        at 0 here. For water (O, H, H) with 6-31G, this would be an array like
+        [0, 0, 0, 0, 0, 1, 1, 2, 2]. , `O` in 6-31G has 5 shells and`H` has two shells.
+    shell_types: np.ndarray (integer)
+        Angular momentum of the shell. Indices start at 0 for 's' orbital, 1 for 'p' etc.
+        For 6-31G for a heavy atom this would be [0, 0, 1, 0, 1] corresponding
+         to [1s, 2s, 2p, 2s, 2p]
+    exponents: np.ndarray (float)
+        Gaussian function decay exponents for the primitives in the basis set.
+    prim_per_shell: np.ndarray (integer)
+        Array denoting the number of primitives per shell. If basis set is 6-31G this will be
+        [6, 3, 3, 1, 1] if the atom is a heavy atom. This corresponds to
+        [1s, 2s, 2p, 2s, 2p]. If additional atoms are present, the array is extended.
+    coeffs: np.ndarray (float)
+        Array of same length as `exponents` containing orbital expansion coefficients.
+    """
+    shells = []
+    counter = 0
+    # First loop over all shells
+    for i, n in enumerate(prim_per_shell):
+        shells.append(Shell(
+            shell_map[i],
+            [abs(shell_types[i])],
+            ['p' if shell_types[i] < 0 else 'c'],
+            exponents[counter:counter + n],
+            coeffs[counter:counter + n][:, np.newaxis]
+        ))
+        counter += n
+    del shell_map
+    del shell_types
+    del prim_per_shell
+    del exponents
+    del coeffs
+
+    obasis = MolecularBasis(tuple(shells), CONVENTIONS, 'L2')
+    return obasis
+
+
+@document_load_one("MWFN", ['atcoords', 'atnums', 'atcorenums', 'energy',
+                            'mo', 'obasis', 'extra', 'title'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    inp = load_mwfn_low(lit)
+
+    # MWFN contains more information than most formats, so the following dict
+    # stores some "extra" stuff.
+    mwfn_dict = {
```

Comment:
Could you use `extra` instead of `mwfn_dict` for clarity. Some things in this dictionary are also present in the other IOData attributes, either directly or indirectly. I'd suggest to take these out. See line 13 of `import this`. :)